modules/keytool/src/main/scala/org/tesselation/keytool/security/Signing.scala

```diff
@@ -1,5 +1,40 @@
 package org.tesselation.keytool.security
 
+import java.security.{PrivateKey, PublicKey, Signature}
+
+import cats.effect.Async
+import cats.syntax.flatMap._
+import cats.syntax.functor._
+
 object Signing {
   val defaultSignFunc = "SHA512withECDSA"
+
+  def signData[F[_]: Async: SecurityProvider](
+    bytes: Array[Byte],
+    signFunc: String = defaultSignFunc
+  )(privateKey: PrivateKey): F[Array[Byte]] =
+    for {
+      s <- Async[F].delay {
+        Signature.getInstance(signFunc, SecurityProvider[F].provider)
+      }
+      _ <- WithSecureRandom.get[F].flatMap { secureRandom =>
+        Async[F].delay { s.initSign(privateKey, secureRandom) }
+      }
+      _ <- Async[F].delay { s.update(bytes) }
+      signed <- Async[F].delay { s.sign() }
+    } yield signed
+
+  def verifySignature[F[_]: Async: SecurityProvider](
+    originalInput: Array[Byte],
+    signedOutput: Array[Byte],
```

Comment:
Isn't `signedOutput` just a signature? If this is the case I would rename the first two parameters of this function to just `inputData` (or just `bytes` like in function above) and `signature`.

Reply:
I left it as it is in the original code